package/MDAnalysis/topology/guessers.py

```diff
@@ -225,6 +226,16 @@ def guess_atom_element(atomname):
         return no_symbols
 
 
+def guess_atom_element_from_mass(mass_guess, tol=0.1):
+    for element, mass in tables.masses.items():
+        if abs(mass - mass_guess) < tol:
+            return element
+    raise ValueError(f'the molecular weight: {mass_guess} amu, is not valid')
+
+
+guess_atom_elements_from_masses = np.vectorize(guess_atom_element_from_mass)
```

Comment:
Whilst vectorize is nice, I believe it does not provide any actual performance improvements instead is just a for loop around the other function.

Since these are user-facing functions, I'd suggest writing a proper function call with docstring that folks can call if they need to.